src/sys/windows_impl.rs

```diff
@@ -160,6 +168,27 @@ impl FileExt for File {
         Ok(())
     }
 
+    fn unset_sparse(&self) -> io::Result<()> {
+        let mut info = FILE_BASIC_INFO {
+            CreationTime: 0,
+            LastAccessTime: 0,
+            LastWriteTime: 0,
+            ChangeTime: 0,
+            FileAttributes: 0,
+        };
+
+        unsafe {
+            SetFileInformationByHandle(
```

Comment:
The linked cow implementation uses `DeviceIoControl`, but we uses `SetFileInformationByHandle`, are you sure they are equivalent?

Reply:
What about this? I thought they were the same, but looks like it doesn't.
```rs
fn unset_sparse(&self) -> io::Result<()> {
        let mut bytes_returned = 0u32;
        let mut sparse_flag: u32 = 0;
    
        unsafe {
            DeviceIoControl(
                self.as_handle(),
                FSCTL_SET_SPARSE,
                Some(&mut sparse_flag as *mut _ as *mut c_void),
                std::mem::size_of_val(&sparse_flag) as u32,
                Option::None,
                0,
                Some(&mut bytes_returned as *mut _),
                Option::None,
            )
        }?;        
    
        Ok(())
    }
```

Reply:
I think you almost got it right:

```
fn unset_sparse(&self) -> io::Result<()> {
        let mut bytes_returned = 0u32;
        let mut sparse_flag: u32 = 0;
    
        unsafe {
            DeviceIoControl(
                self.as_handle(),
                FSCTL_SET_SPARSE,
                Some(&mut sparse_flag as *mut _ as *mut c_void),
                mem::size_of::<u32>() as u32,
                None,
                0,
                Some(&mut bytes_returned as *mut _),
                None,
            )
        }?;        
    
        Ok(())
    }
```